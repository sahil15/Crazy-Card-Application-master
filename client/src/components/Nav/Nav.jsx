import React from "react";
import { ThemeConsumer } from "../../contexts/theme";
import { makeStyles } from "@material-ui/core/styles";
import AppBar from "@material-ui/core/AppBar";
import Toolbar from "@material-ui/core/Toolbar";
import Typography from "@material-ui/core/Typography";
import Button from "@material-ui/core/Button";
import IconButton from "@material-ui/core/IconButton";
import MenuIcon from "@material-ui/icons/Menu";


const useStyles = makeStyles((theme) => ({
  root: {
    flexGrow: 1,
  },
  menuButton: {
    marginRight: theme.spacing(2),
  },
  title: {
    flexGrow: 1,
  },
  appBar: {
    marginTop: 10,
    marginBottom: 10
  },
  btn: {
      fontSize: 30
  }
}));
export default function Nav() {
  const classes = useStyles();
  return (
    <ThemeConsumer>
      {({ theme, toggleTheme }) => (
        <>
          <AppBar position="static" className={classes.appBar}>
            <Toolbar>
              <IconButton
                edge="start"
                className={classes.menuButton}
                color="inherit"
                aria-label="menu"
              >
                <MenuIcon />
              </IconButton>
              <Typography variant="h6" className={classes.title}>
                Crazy Card Application
              </Typography>
              <Button
                className={classes.btn}
                color="inherit"
                onClick={toggleTheme}
              >
                {theme === "light" ? "ðŸ”¦" : "ðŸ’¡"}
              </Button>
            </Toolbar>
          </AppBar>
        </>

      )}
    </ThemeConsumer>
  );
}
