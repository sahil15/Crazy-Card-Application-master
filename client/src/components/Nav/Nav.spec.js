import React from "react";
import { shallow } from "enzyme";
import Nav from "./Nav";

describe("Tests for Nav menu", () => {
  
  it("Renders correctly", () => {
    const wrapper = shallow(
      <Nav/>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
