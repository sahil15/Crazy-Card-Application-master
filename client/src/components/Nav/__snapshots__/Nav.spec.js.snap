// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Nav menu Renders correctly 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
